app/celery/nightly_tasks.py

```diff
@@ -409,3 +412,48 @@ def pick_bst_quarter_to_delete_back_from(oldest_notification_bst):
         deletion_date = deletion_date + dateutil.relativedelta.relativedelta(months=3)
 
     return deletion_date
+
+
+@notify_celery.task(name="delete_unneeded_notification_history_by_hour")
+def delete_unneeded_notification_history_by_hour():
```

Comment:
What's our plan for triggering this? A manual step, or something else?

Reply:
The plan is to kick this task off once manually by logging into production and creating a task. 

I did think about scheduling a task for a particular day of the year but it felt a bit ick and I thought there was a chance it would get left in the code (not to mention picking a day to schedule it for that is as soon as possible but is also after the time it takes to get this reviewed and approved).

Open to options though.

Reply:
I think the only reason I ask is that, in the case it's a manual thing, maybe we want the option to be able to provide a smaller daterange in the beginning as a test run. As it stands this will generate all of the hours across the entire period, so if something goes wrong from the get go then it's a bit more cleanup.

But I suspect it's probably manageable.

As a test run we could just manually log some of the child tasks with our own hour ranges.

And then later if it goes wrong we can just scale down the broadcast worker and purge the queue.

So probably not worth changing it after all.

Reply:
I'm happy you've talked yourself out of it :) It would be fairly quick to scale down the broadcasts worker if we want to stop things.